app/lib/features/segment_deletion_config.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Features
+  class SegmentDeletionConfig
+    def self.configure(config = {})
+      @configuration = new(config)
+    end
+
+    def self.configuration
+      (@configuration || configure)
+    end
+
+    def self.config
+      configuration.config
+    end
+
+    def self.enabled?
+      configuration.enabled?
+    end
+
+    def initialize(config = {})
+      format_config(config)
+      check_validity(config)
+    end
+
+    def self.valid_environment?(environment: Rails.env)
+      # CAUTION!:
+      # This is because this feature destroys real data unless the call to segment is stubbed and without real credentials in the config (test environment),
+      # so we prevent this feature to be used in preview or development thinking that be just checking around while actually destroying real data by accident
+      %w[production test].include?(environment)
+    end
+
+    def self.log_disabled_warning
+      warn_message = '[WARNING] Can\'t enable "deletion of users in segment.'
```

Comment:
Note, if we can achieve separating environments, this and the likes are not needed anymore

Reply:
@hallelujah I have no idea what you mean by _the likes_, but it is true this is only needed for the environments problem 😉